### PR TITLE
UCT/BASE: Move uct_worker code from uct_md.{h,c} to uct_worker.{h,c}

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3729/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1792/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1795/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3732/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3737/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
@shamisp pls review

File: src/uct/base/uct_worker.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
```
Comment (shamisp) [inline src/uct/base/uct_worker.c]: you moved the code without preserving copyrights. Please copy-n-paste copyright from the original file

File: src/uct/base/uct_worker.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
```
Comment (shamisp) [inline src/uct/base/uct_worker.h]: the same here

File: src/uct/base/uct_worker.c (hunk 1 above)
Comment (yosefe) [inline src/uct/base/uct_worker.c]: fixed

File: src/uct/base/uct_worker.h (hunk 2 above)
Comment (yosefe) [inline src/uct/base/uct_worker.h]: this one does not need extra copyright since it's copied from uct_md.h which has only this one

**Comment by yosefe** [comment]
@shamisp pls take a look again

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1813/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3753/ for details (Mellanox internal link).

**Pull request merged**